swift/DataDonationExample/Services/DataDonationSDKService.swift

```diff
@@ -23,7 +23,11 @@ final class DataDonationSDKService {
     private let client: DataDonationClientProtocol!
     init(userSessionProvider: UserSessionTokenProviderProtocol) {
         self.userSessionProvider = userSessionProvider
-        self.client = Client.Factory().getInstance(environment: .staging, userSession: userSessionProvider)
+        self.client = Client.Factory().getInstance(
+            environment: .staging,
+            userSession: userSessionProvider,
+            coroutineScope: nil
```

Comment:
did you test out of curiousity what happens if adding main coroutine scope here? 
also we might need a way to create some scopes since this is now public

Reply:
Not yet, since unless iOS is capable of generating one (which I think it is not at the moment), we do not run into this situation. Also please consider, with Kotlin 1.5.x we are capable of hiding this a bit more, since it let us use default parameter on expect/actual things. Also at moment, I would not advise to give iOS the possibility to produce the scope unless we have a proper documentation for that, so we make transparent what are the current boundaries are.